### PR TITLE
BE - Exclusions - Share my information - PDF - Exclusions

### PR DESCRIPTION
**Pull request by dbgoaco**
[DP-447](https://noticingsystem.atlassian.net/browse/DP-447)
- includes form sections: Exclusion, Financial Information, Qualifications and Trade assurances

### CODE REVIEW

File: Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs [hunk 1]
```diff
@@ -10,21 +10,90 @@ namespace CO.CDP.DataSharing.WebApi.DataService;
...
+    private const string FinancialInformationSectionName = "Financial Information";
+    private const string QualificationsSectionName = "Qualifications";
+    private const string TradeAssurancesSectionName = "Trade assurances";
+
     public async Task<SharedSupplierInformation> GetSharedSupplierInformationAsync(string shareCode)
     {
         var sharedConsent = await shareCodeRepository.GetByShareCode(shareCode)
                             ?? throw new ShareCodeNotFoundException(Constants.ShareCodeNotFoundExceptionMessage);
 
+        var exclusions = sharedConsent.AnswerSets.Where(a => a.Section.Title == ExclusionsSectionName).ToArray();
```
Comment (dharmverma) [inline Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs:23]: Shouldn't we generalize forms implementation in pdf?
We can excluded ShareCode using flag Declaration type

Comment (dbgoaco) [reply Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs]: Not sure what you mean. I have generalized it such that all form sections are now output in the PDF except declarations.

Comment (dharmverma) [reply Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs]: Cool what you did is what I meant :)

File: Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs [hunk 2]
```diff
@@ -10,21 +10,81 @@ namespace CO.CDP.DataSharing.WebApi.DataService;
...
 public class DataService(IShareCodeRepository shareCodeRepository, IConnectedEntityRepository connectedEntityRepository) : IDataService
 {
+    private const string DeclarationsSectionName = "Declarations";
+
     public async Task<SharedSupplierInformation> GetSharedSupplierInformationAsync(string shareCode)
     {
         var sharedConsent = await shareCodeRepository.GetByShareCode(shareCode)
                             ?? throw new ShareCodeNotFoundException(Constants.ShareCodeNotFoundExceptionMessage);
-
+        var allFormSectionsExceptDeclaractions = sharedConsent.AnswerSets.Where(a => a.Section.Title != DeclarationsSectionName);
```
Comment (dharmverma) [inline Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs:19]: Can improve with below code, this is what my earlier comment was pointing too.
```
var allFormSectionsExceptDeclaractions = sharedConsent.AnswerSets.Where(a =>
    a.Section.Type != OrganisationInformation.Persistence.Forms.FormSectionType.Declaration);
```

Comment (dbgoaco) [reply Services/CO.CDP.DataSharing.WebApi/DataService/DataService.cs]: Ah yes ok. I wasn't aware of that constant. I've updated the code as you've suggested.

**Review by rmohammed-goaco — APPROVED** [review]

**Pull request merged**